src/sidebar/services/frame-sync.js

```diff
@@ -226,7 +234,7 @@ export class FrameSyncService {
   /**
    * Connect to the host frame and guest frame(s) in the current browser tab.
    */
-  connect() {
+  async connect() {
```

Comment:
Because `connect` is now async, any failures will no longer trigger the `catch` block at the bottom of `src/sidebar/index.js` and thus the exception won't be reported via `reportLaunchError`. I think fixing that would be too big of a change for this PR, but we should address it in future.